plugins/module_utils/template.py

```diff
@@ -207,3 +207,25 @@ def get_l3out_node_routing_policy_object(self, uuid=None, name=None, fail_module
                 "L3Out Node Routing Policy", existing_l3out_node_routing_policy, [KVPair("uuid", uuid) if uuid else KVPair("name", name)], fail_module
             )
         return existing_l3out_node_routing_policy  # Query all objects
+
+    def get_l3out_interface_routing_policy_object(self, uuid=None, name=None, fail_module=False):
+        """
+        Get the L3Out Interface Routing Policy by UUID or Name.
+        :param uuid: UUID of the L3Out Interface Routing Policy to search for -> Str
+        :param name: Name of the L3Out Interface Routing Policy to search for -> Str
+        :param fail_module: When match is not found fail the ansible module -> Bool
+        :return: Dict | None | List[Dict] | List[]: The processed result which could be:
+                 When the UUID | Name is existing in the search list -> Dict
+                 When the UUID | Name is not existing in the search list -> None
+                 When both UUID and Name are None, and the search list is not empty -> List[Dict]
+                 When both UUID and Name are None, and the search list is empty -> List[]
+        """
+        existing_l3out_interface_routing_policy = self.template.get("tenantPolicyTemplate", {}).get("template", {}).get("l3OutIntfPolGroups", [])
+        if uuid or name:  # Query a specific object
+            return self.get_object_by_key_value_pairs(
+                "L3Out Interface Routing Policy",
+                existing_l3out_interface_routing_policy,
+                [KVPair("uuid", uuid) if uuid else KVPair("name", name)],
+                fail_module,
+            )
+        return existing_l3out_interface_routing_policy  # Query all objects
```

Comment:
Will this function be used in more than one module? Otherwise, we should move it to ndo_l3out_interface_routing_policy module.

Reply:
Yes, it will be used for the L3Out Template Interface Group Policy module.